techlibs/xilinx/synth_xilinx.cc

```diff
@@ -450,14 +460,13 @@ struct SynthXilinxPass : public ScriptPass
 		}
 
 		if (check_label("fine")) {
-			run("dffsr2dff");
-			run("dff2dffe");
+			// Required for dff2dffe to work well.
+			run("clean");
```

Comment:
`clean` calls can be quite expensive. Can you share what this call cleans up to help `dff2dffe` do its job more efficiently?

Reply:
When `dff2dffs` folds a mux with a const input into the FF as a sync set/reset, it leaves the original mux still there, unused. When `dff2dffe` runs, it notices that the second mux that would have been folded into CE has more than one user (the now-dead set/reset mux), and will not perform the conversion.

Reply:
Have you considered putting `clean` call after `dff2dffs` so that `-run fine:` doesn't start with a clean?

Reply:
Wait, wouldn't https://github.com/YosysHQ/yosys/blob/c5f2a530c4f3b661508fe1c96d928a252628fc68/techlibs/xilinx/synth_xilinx.cc#L454-L458 already run `opt_clean`?

Reply:
Good point, removed the clean call.